### PR TITLE
feat(source): Add source v2

### PR DESCRIPTION
**Pull request by raulb**
Fixes https://github.com/meroxa/meroxa-go/issues/162

It also refactors some some variable and method names that uses the V2 path to make them all consistent.

### CODE REVIEW

File: pkg/meroxa/source.go [hunk 1]
```diff
@@ -6,19 +6,47 @@ import (
...
+type CreateSourceInputV2 struct {
+	Environment *EntityIdentifier `json:"environment,omitempty"`
+}
+
 type SourceBlob struct {
 	Url string `json:"url"`
 }
 
+// CreateSource using the v1 path won't accept body parameters, and it's an unauthenticated request
 func (c *client) CreateSource(ctx context.Context) (*Source, error) {
```
Comment (dianadoherty) [inline pkg/meroxa/source.go:26]: Sources v1 and v2 look identical. Is there a need for us to have these separate endpoints?

Comment (raulb) [reply pkg/meroxa/source.go:26]: @dianadoherty they aren't identical. v1 doesn't accept body parameters whereas v2 does.

**Review by anna-cross — APPROVED** [review]
These make sense to me

File: pkg/meroxa/source.go (hunk 1 above)
Comment (dianadoherty) [reply pkg/meroxa/source.go:26]: Oh gotcha, did not notice this! Looks good~

**Review by dianadoherty — APPROVED** [review]

File: pkg/meroxa/source.go [hunk 2]
```diff
@@ -6,19 +6,47 @@ import (
 	"net/http"
 )
 
-const sourcesBasePath = "/v1/sources"
+const sourcesBasePathV1 = "/v1/sources"
+const sourcesBasePathV2 = "/v2/sources"
 
 type Source struct {
```
Comment (anna-cross) [inline pkg/meroxa/source.go:12]: Do we want to return Environment too from platform CreateSourceV2 to CLI?

Comment (raulb) [reply pkg/meroxa/source.go:12]: @anna-cross I don't think we need it. We only need both `put_url` and `get_url` from the CLI to continue with updating the source and creating the build later on.

**Pull request merged**